src/main/drivers/adc_stm32h7xx.c

```diff
@@ -322,24 +328,30 @@ void adcInit(const adcConfig_t *config)
             // Found a tag map entry for this input pin
             // Find an ADC device that can handle this input pin
 
-            for (dev = 0; dev < ADCDEV_COUNT; dev++) {
-                if (!adcDevice[dev].ADCx
+            bool useConfiguredDevice = (dev != ADCINVALID) && (adcTagMap[map].devices & (1 << dev));
+
+            if (!useConfiguredDevice) {
+                // If the ADC was configured to use a specific device, but that device was not active, then try and find another active instance that works for the pin.
+
+                for (dev = 0; dev < ADCDEV_COUNT; dev++) {
+                    if (!adcDevice[dev].ADCx
 #ifndef USE_DMA_SPEC
-                     || !adcDevice[dev].dmaResource
+                         || !adcDevice[dev].dmaResource
```

Comment:
👾

Reply:
what does that emoji mean?  Is there something wrong with the indentation?  previously there was only a one-space indentation on a line that should have had more indentation.

Reply:
Yeah should be 4 spaces 👾

Reply:
done.